tests/entrypoints/openai/test_metrics.py

```diff
@@ -84,7 +104,9 @@ async def client(server):
 
 @pytest.mark.asyncio
 async def test_metrics_counts(server: RemoteOpenAIServer,
-                              client: openai.AsyncClient):
+                              client: openai.AsyncClient, use_v1: bool):
+    if use_v1:
+        pytest.skip("Skipping test on vllm V1")
```

Comment:
🤔 Won't this spend the time to build the `server` fixture before just skipping the test?

We could use the `run_with_both_engines` fixture so that `@skip_v1` can be used here, and do 
```
for metric in (EXPECTED_METRICS_V1 if os.getenv('VLLM_USE_V1') == "1" else EXPECTED_METRICS):
```
in the next test where we need to split the logic for v1. It's not nearly as clean but that would let us skip the tests during collection instead of at runtime after fixture setup

Reply:
Thanks for the review, Joe

The `server` fixture is a "module-level" fixture - it gets initialized once and shared across the three tests

So the v1 server will always be used by one test, even though the other two gets skipped

And because `server` is a module-level fixture, we can not use function-level fixtures like `run_with_both_engines` or `monkeypatch`

Nor can a function-level decorator cause a module-level decorator to skip a specific test function

I went the `run_with_both_engines` route initially and fumbled my way into the above understanding of the constraints

Reply:
Oh yeah you're right, I missed the server definition in this file 👍